xarray/backends/zarr.py

```diff
@@ -1218,64 +1218,85 @@ def open_datatree(
         zarr_version=None,
         **kwargs,
     ) -> DataTree:
-        from xarray.backends.api import open_dataset
         from xarray.core.datatree import DataTree
+
+        filename_or_obj = _normalize_path(filename_or_obj)
+        groups_dict = self.open_groups_as_dict(filename_or_obj, **kwargs)
+
+        return DataTree.from_dict(groups_dict)
+
+    def open_groups_as_dict(
+        self,
+        filename_or_obj: str | os.PathLike[Any] | BufferedIOBase | AbstractDataStore,
+        *,
+        mask_and_scale=True,
+        decode_times=True,
+        concat_characters=True,
+        decode_coords=True,
+        drop_variables: str | Iterable[str] | None = None,
+        use_cftime=None,
+        decode_timedelta=None,
+        group: str | Iterable[str] | Callable | None = None,
+        mode="r",
+        synchronizer=None,
+        consolidated=None,
+        chunk_store=None,
+        storage_options=None,
+        stacklevel=3,
+        zarr_version=None,
+        **kwargs,
+    ) -> dict[str, Dataset]:
+
         from xarray.core.treenode import NodePath
 
         filename_or_obj = _normalize_path(filename_or_obj)
+
+        # Check for a group and make it a parent if it exists
         if group:
             parent = NodePath("/") / NodePath(group)
-            stores = ZarrStore.open_store(
-                filename_or_obj,
-                group=parent,
-                mode=mode,
-                synchronizer=synchronizer,
-                consolidated=consolidated,
-                consolidate_on_close=False,
-                chunk_store=chunk_store,
-                storage_options=storage_options,
-                stacklevel=stacklevel + 1,
-                zarr_version=zarr_version,
-            )
-            if not stores:
-                ds = open_dataset(
-                    filename_or_obj, group=parent, engine="zarr", **kwargs
-                )
-                return DataTree.from_dict({str(parent): ds})
         else:
             parent = NodePath("/")
-            stores = ZarrStore.open_store(
-                filename_or_obj,
-                group=parent,
-                mode=mode,
-                synchronizer=synchronizer,
-                consolidated=consolidated,
-                consolidate_on_close=False,
-                chunk_store=chunk_store,
-                storage_options=storage_options,
-                stacklevel=stacklevel + 1,
-                zarr_version=zarr_version,
-            )
-        ds = open_dataset(filename_or_obj, group=parent, engine="zarr", **kwargs)
-        tree_root = DataTree.from_dict({str(parent): ds})
+
+        stores = ZarrStore.open_store(
+            filename_or_obj,
+            group=parent,
```

Comment:
`parent` at this point is a `NodePath` object - perhaps better to coerce it to `str` first?

Reply:
(resolved in https://github.com/pydata/xarray/pull/9469/commits/dac8c326eb90f88463c9808fdd70e0815bb40259)